### ISSUE TITLE
CI - separate build and test stages

### ISSUE THREAD

**Issue by AlexSkrypnyk** [drevops/vortex#782]
With using docker, it is possible to build images and cache them based on the hash of `composer.json` and other files like this to store the image in the cache in one job to pick it up in another job

**Comment by AlexSkrypnyk** [comment]
The default limit of the concurrent jobs in Circle CI was raised to 30 for orgs. This means that it is no possible to have multiple "fast" jobs.

**Comment by AlexSkrypnyk** [comment]
The caching size between builds is limited by the size of the produced images. The time it takes to cache and restore Docker images between builds may be longer than the build of the new images.

Besides, the built images will not be shareable between jobs. To achieve this, each image would have to be put in own cache using a separate cache key. This will also mean that an additional storage would need to be provisioned.

Basically, while technically possible, it is too complex to implement and would not get that many gains for a generic project.
